File: Core/V1/Module/Github/Model/Workflow.php
<?php

namespace Core\Github\Model;

class Workflow
{
    private string $ref;

    /** @var array<string, string> */
    private array $inputs;

    /**
     * Workflow constructor.
     *
     * @param string $ref
     * @param array<string, string> $inputs
     */
    public function __construct(string $ref, array $inputs)
    {
        $this->ref = $ref;
        $this->inputs = $inputs;
    }

    /**
     * Get the ref.
     *
     * @return string
     */
    public function getRef(): string
    {
        return $this->ref;
    }

    /**
     * Get the inputs.
     *
     * @return array<string, string>
     */
    public function getInputs(): array
    {
        return $this->inputs;
    }

    /**
     * Serialize the Workflow object to JSON.
     *
     * @return string
     */
    public function serialize(): string
    {
        return json_encode([
            'ref' => $this->ref,
            'inputs' => $this->inputs,
        ]);
    }
}

File: Core/Lib/nanorest/src/main/java/org/sindria/xpipe/core/lib/nanorest/github/model/Workflow.java
package org.sindria.xpipe.core.lib.nanorest.github.model;

import org.json.JSONObject;
import org.sindria.xpipe.core.lib.nanorest.serializer.JsonSerializer;

import java.util.HashMap;

public class Workflow {

    private final String ref;

    private final HashMap<String, String> inputs;

    public Workflow(String ref, HashMap<String, String> inputs) {
        this.ref = ref;
        this.inputs = inputs;
    }

    public String getRef() {
        return ref;
    }

    public HashMap<String, String> getInputs() {
        return inputs;
    }

    public JSONObject serialize() {
        return new JSONObject(JsonSerializer.toJson(this));
    }
}
